### PR TITLE
ztp: CNF-7798: git-first-search-path-for-ztp

### PR DESCRIPTION
**Pull request by sabbir-47**
source crs are looked first at git repo, then inside the ztp container

/cc @serngawy @imiller0

### CODE REVIEW

File: ztp/policygenerator/utils/filesHandler.go [hunk 1]
```diff
@@ -55,24 +55,35 @@ func (fHandler *FilesHandler) ReadSourceFile(fileName string) ([]byte, error) {
...
 	if err != nil {
 		return nil, err
 	}
-	dir = filepath.Dir(ex)
-	ret, err = fHandler.ReadFile(dir + "/" + SourceCRsPath + "/" + fileName)
+	sourceCRPathGit := gitDir + "/" + SourceCRsPath + "/" + fileName
 
-	// added fail safe for test runs as `os.Executable()` will fail for tests
+	// path of the local executable
+	localDir, err = os.Executable()
```
Comment (serngawy) [inline ztp/policygenerator/utils/filesHandler.go:73]: I believe we should read first the `os.ReadFile(sourceCRPathGit)`  if it fail to read the file then read local `os.Executable()`

Comment (sabbir-47) [reply ztp/policygenerator/utils/filesHandler.go:73]: This is exactly what we are doing at line 80 below.

File: ztp/policygenerator/utils/filesHandler.go [hunk 2]
```diff
@@ -55,24 +55,35 @@ func (fHandler *FilesHandler) ReadSourceFile(fileName string) ([]byte, error) {
...
+	sourceCRPathLocal := dir + "/" + SourceCRsPath + "/" + fileName
+
+	fileByte, err = os.ReadFile(sourceCRPathGit)
+	if errors.Is(err, os.ErrNotExist) {
+		fileByte, err := os.ReadFile(sourceCRPathLocal)
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, errors.New(fileName + " is not found both in Git path: " + sourceCRPathGit + " and in the ztp container path: " + sourceCRPathLocal)
 		}
-		ret, err = fHandler.ReadFile(dir + "/" + SourceCRsPath + "/" + fileName)
+		return fileByte, nil
```
Comment (imiller0) [inline ztp/policygenerator/utils/filesHandler.go:86]: By returning nil here we are masking any other potential error cases (eg file permissions) from the os.ReadFile. Any others are very unlikely but it seems like we should use:
```suggestion
		return fileByte, err
```

**Review by imiller0 — COMMENTED** [review]
/lgtm

**Review by serngawy — APPROVED** [review]

**Pull request merged**